### DISCUSSION TITLE
bufferline wrapping

### DISCUSSION THREAD

**Discussion by adamsky** [helix-editor/helix#10199] [Ideas]
Working with 5 or more buffers on a small screen I often found myself wanting the ability to wrap the bufferline into a second or even third line instead of having it go "off screen".

I couldn't find a similar enhancement suggestion already mentioned. I have a working solution and I'm already using it locally. Wondering if this is something folks would like to see at all. If so should this be the new default bufferline behavior or perhaps hidden under an additional config property?

Here's an example implementation I'm currently using https://github.com/helix-editor/helix/pull/8324

**Comment by aslowwriter** [comment]
I often have a split screen open (screen is split by multiplexer not helix) and so often I can't see which buffer I"m on if they're near the end of the list. I would love to see either this, or parhaps a "scrolling" bufferline, where it moves the start off the bufferline off screen as you cycle through buffers

**Comment by watsoncj** [comment]
Remember `Space+b` opens the buffer picker which is probably a better option if you have many buffers open.

Most editors (and browsers) have a mode where just the surrounding tabs are shown. That might be preferred to wrapping as it saves space and still gives you the relative buffers.

Here's a quick example of bufferline wrapping: https://github.com/watsoncj/helix/commit/cb759d9fa28bc703292435ade727f085375449f1

- grows without bounds
- has an issue with truncating the last file on a line
- lacks config to disable 

![Screenshot from 2024-04-10 14-02-14](https://github.com/helix-editor/helix/assets/31694/b162d02c-7413-4f98-b1db-9f205cc02a5c)
![Screenshot from 2024-04-10 14-02-56](https://github.com/helix-editor/helix/assets/31694/5a0140c4-3058-44bd-9c47-f06866bb7fb0)

**Comment by James-Firth** [comment]
I'm not sure if I want it to wrap, but I would like the currently active buffer to be visible, once I go "off screen" with `buffer-next` I can't tell where I am any more.

So giving a 👍 since I think this is a useful discussion!

**Comment by jpeeler** [comment]
Vim's (v9.1.0452) behavior seems to have the same off screen issue, except has some heuristics for shortening each file upon each new addition such that a lot more can fit. Eventually gives up once each file name is four characters long. (Edit: sorry took the screenshot too soon, it does go down to 4 chars.)

![image](https://github.com/user-attachments/assets/d8f272c5-a5e0-48a3-a8a6-3478574c88ee)

```
:buffers
  1  a   "[No Name]"                    line 1
  2  a   "asdfasdlfkjasdlfkj1"          line 0
  3  a   "asdflkjasdlfkjasdflkj2"       line 0
  4  a   "asldfkjasldfkjasdlfkj3"       line 0
  5  a   "asdlfkjasdflkjasdflkj4"       line 0
  6  a   "asdlfkjasdlfkjasdflkja5"      line 0
  7  a   "asdflkjasdflkajsdflkjasd6"    line 0
  8  a   "asdlfkjasdflkjasdlfkj7"       line 0
  9  a   "asdlfkjasdlfkjasdlfkj8"       line 0
 10  a   "asdflkjasdflkjasdlfkjasldkfjasldkfj9" line 0
 11  a   "asldfkjasdlfkjasdlfkjas10"    line 0
 12  a   "asdlfkjasdflkjasdflkajsdd11"  line 0
 13  a   "asdflkjasdflkajsdflkj12"      line 0
 14  a   "aalsdkfjasldkfjasdlfkjasdflkj13" line 0
 15  a   "asdlfkjasdlfkjasdflkjasdf14"  line 0
 16 #a   "asdflkjasdflkjasdflaskdjf15"  line 0
 17 %a   "asdflkjasdflkajsdfasdlkj16"   line 1
Press ENTER or type command to continue
```

After a bunch of deletions:

```
:buffers
  1  a   "[No Name]"                    line 1
  2  a   "asdfasdlfkjasdlfkj1"          line 0
  3  a   "asdflkjasdlfkjasdflkj2"       line 0
  4  a   "asldfkjasldfkjasdlfkj3"       line 0
  5  a   "asdlfkjasdflkjasdflkj4"       line 0
  6  a   "asdlfkjasdlfkjasdflkja5"      line 0
  7  a   "asdflkjasdflkajsdflkjasd6"    line 0
  8 #a   "asdlfkjasdflkjasdlfkj7"       line 0
  9 %a   "asdlfkjasdlfkjasdlfkj8"       line 1
Press ENTER or type command to continue
```
![image](https://github.com/user-attachments/assets/53b378e4-0f85-4860-8bec-300f9c461037)

**Comment by daedroza** [comment]
I am surprised that something basic like this is not present in helix.
